lib/chromic_pdf/pdf/chrome_runner.ex

```diff
@@ -1,25 +1,66 @@
 # SPDX-License-Identifier: Apache-2.0
 
-defmodule ChromicPDF.ChromeImpl do
+defmodule ChromicPDF.ChromeRunner do
   @moduledoc false
 
-  @behaviour ChromicPDF.Chrome
+  @behaviour ChromicPDF.Connection.ChromeRunner
 
+  @impl true
   def spawn(opts) do
-    port = Port.open({:spawn, chrome_command(opts)}, port_opts(opts))
+    port_opts = append_if([:binary], :nouse_stdio, !discard_stderr?(opts))
+    port_cmd = shell_command("--remote-debugging-pipe", opts)
+
+    port = Port.open({:spawn, port_cmd}, port_opts)
     Port.monitor(port)
 
     {:ok, port}
   end
 
+  @impl true
   def send_msg(port, msg) do
     send(port, {self(), {:command, msg <> "\0"}})
 
     :ok
   end
 
-  # --------------------------- Chrome Command Line ----------------------------------
-  #
+  @spec warmup(keyword()) :: :ok
+  def warmup(opts) do
+    "<html><head></head><body></body></html>\n" =
```

Comment:
```suggestion
    "<html>" <> _ =
```
does it matter what is printed? Seems fragile. When chrome decides to remove the additional `\n` from `about:blank` then this would break, right?

Reply:
Just noticed it's actually broken when using `discard_stderr: false`

```
iex(1)> ChromicPDF.warm_up(discard_stderr: false)
** (MatchError) no match of right hand side value: "[1214/100758.006638:WARNING:bluez_dbus_manager.cc(247)] Floss manager not present, cannot set Floss enable/disable.\n<html><head></head><body></body></html>\n"
    (chromic_pdf 1.4.0) lib/chromic_pdf/pdf/chrome_runner.ex:28: ChromicPDF.ChromeRunner.warm_up/1
```

Reply:
This is actually pretty surprising IMO

```elixir
iex(1)> :os.cmd(String.to_charlist("echo foo >&2"))
'foo\n'
```

Docs on [`os.cmd`](https://www.erlang.org/doc/man/os.html#cmd-1) say 

> Executes Command in a command shell of the target OS, captures the standard output of the command, and returns this result as a string.

... which would also be my intuition for its behaviour. But nope, it also captures `stderr` :shrug:

Reply:
See also http://erlang.org/pipermail/erlang-questions/2016-June/089640.html

Reply:
changed this around a bit, removing the stdout output (TBH I don't think `about:blank` is going to change much in terms of markup) and returning the capture stderr